neurox/interpretation/gaussian_probe.py

```diff
@@ -0,0 +1,282 @@
+                    )
+                )
+
+    def _compute_probs(self, selected_features, set_name):
+        features = self.feature_sets[set_name]
+        with torch.no_grad():
+            log_probs = []
+            for i in range(self.labels_dim):
+                if self.train_categorical[i] == 0:
+                    log_probs.append(torch.zeros(features.shape[0], dtype=torch.double))
+                else:
+                    log_probs.append(
+                        self.distributions[i].log_prob(features[:, selected_features])
+                    )
+            log_probs = torch.stack(log_probs, dim=1)
+            log_prob_times_cat = log_probs + self.train_categorical.log()
+            self.not_normalized_probs = log_prob_times_cat
+            self.normalizer = log_prob_times_cat.logsumexp(dim=1)
+        self.probs = log_prob_times_cat - self.normalizer.unsqueeze(1)
+
+    def _predict(self, set_name: str):
+        preds = self.probs.argmax(dim=1)
+        labels = self.label_sets[set_name]
+        accuracy = ((preds == labels)).nonzero(as_tuple=False).shape[0] / labels.shape[
+            0
+        ]
+        categorical = self.categorical_sets[set_name]
+        entropy = torch.distributions.Categorical(categorical).entropy()
+        with torch.no_grad():
+            conditional_entropy = -self.probs[
+                list(range(self.probs.shape[0])), labels
+            ].mean()
+        mutual_inf = (entropy - conditional_entropy) / torch.tensor(2.0).log()
+        return preds, labels, accuracy, mutual_inf.item(), (mutual_inf / entropy).item()
+
+
+def train_probe(X, y):
+    """
+    Train a Gaussian probe.
+    This method trains a linear classifier that can be used as a probe to perform
+    neuron analysis. Use this method when the task that is being probed for is a
+    classification task. A logistic regression model is trained with Cross
+    Entropy loss. The optimizer used is Adam with default ``torch.optim``
+    package hyperparameters.
+    Parameters
+    ----------
+    X_train : numpy.ndarray
+        Numpy Matrix of size [``NUM_TOKENS`` x ``NUM_NEURONS``]. Usually the
+        output of ``interpretation.utils.create_tensors``. ``dtype`` of the
+        matrix must be ``np.float32``
+    y_train : numpy.ndarray
+        Numpy Vector with 0-indexed class labels for each input token. The size
+        of the vector must be [``NUM_TOKENS``].  Usually the output of
+        ``interpretation.utils.create_tensors``. Assumes that class labels are
+        continuous from ``0`` to ``NUM_CLASSES-1``. ``dtype`` of the
+        matrix must be ``np.int``
+    Returns
+    -------
+    probe : interpretation.linear_probe.LinearProbe
+        Trained probe for the given task.
+    """
+    return GaussianProbe(X, y)
+
+
+def evaluate_probe(
+    probe,
+    X_test,
+    y_test,
+    metric="accuracy",
+    return_predictions=False,
+    selected_neurons=list(np.arange(768)),
```

Comment:
We should not have the `768` constant here, because some models have more (or less neurons in a layer). Rather just set it to "-1" or "None", and in the code check for this and set it to however many neurons you have in `X_test`